src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,89 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Commands;
+
+namespace NuGet.CommandLine
+{
+    [Command(
+        typeof(NuGetCommand),
+        "search",
+        "SearchCommandDescription",
+        UsageSummaryResourceName = "SearchCommandUsageSummary",
+        UsageDescriptionResourceName = "SearchCommandUsageDescription",
+        UsageExampleResourceName = "SearchCommandUsageExamples")]
+    public class SearchCommand : Command
+    {
+        private readonly List<string> _sources = new List<string>();
+
+        [Option(typeof(NuGetCommand), "ListCommandSourceDescription")]
+        public ICollection<string> Source
+        {
+            get { return _sources; }
+        }
+
+        [Option(typeof(NuGetCommand), "ListCommandVerboseListDescription")]
+        public bool Verbose { get; set; }
+
+        [Option(typeof(NuGetCommand), "ListCommandAllVersionsDescription")]
+        public bool AllVersions { get; set; }
+
+        [Option(typeof(NuGetCommand), "ListCommandPrerelease")]
+        public bool Prerelease { get; set; }
+
+        [Option(typeof(NuGetCommand), "ListCommandIncludeDelisted")]
+        public bool IncludeDelisted { get; set; }
+
+        private IList<Configuration.PackageSource> GetEndpointsAsync()
+        {
+            var configurationSources = SourceProvider.LoadPackageSources()
+                .Where(p => p.IsEnabled)
+                .ToList();
+
+            IList<Configuration.PackageSource> packageSources;
+            if (Source.Count > 0)
+            {
+                packageSources = Source
+                    .Select(s => Common.PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
+                    .ToList();
+            }
+            else
+            {
+                packageSources = configurationSources;
+            }
+            return packageSources;
+        }
+
+        public async override Task ExecuteCommandAsync()
+        {
+            if (Verbose)
+            {
+                Console.WriteWarning(LocalizedResourceManager.GetString("Option_VerboseDeprecated"));
+                Verbosity = Verbosity.Detailed;
+            }
+
+            var listCommandRunner = new ListCommandRunner();
+            var listEndpoints = GetEndpointsAsync();
+
+            var list = new ListArgs(Arguments,
+                listEndpoints,
+                Settings,
+                Console,
+                Console.PrintJustified,
+                Verbosity == Verbosity.Detailed,
+                LocalizedResourceManager.GetString("ListCommandNoPackages"), 
+                LocalizedResourceManager.GetString("ListCommand_LicenseUrl"),
+                LocalizedResourceManager.GetString("ListCommand_ListNotSupported"),
+                AllVersions, 
+                IncludeDelisted,
+                Prerelease,
+                CancellationToken.None);
+
+            await listCommandRunner.ExecuteCommand(list);
+        }
```

Comment:
To me, it makes sense for `nuget list` to return all results it can possibly find. However, it also seems like `nuget search` should only return the most relevant results. Can we add a limit?

Reply:
//cc @anangaur